### ISSUE TITLE
azureblob: server side copy too slow

### ISSUE THREAD

**Issue by ncw** [rclone/rclone#8249]
The rclone [azureblob backend](https://rclone.org/azureblob/) currently offers serve side copies using the [Copy Blob API](https://learn.microsoft.com/en-us/rest/api/storageservices/copy-blob). This copies blobs in a single threaded way on the server. Rclone has to poll the server for completion. It turns out this API is quite slow and doesn't scale well. This causes rclone to be much slower than azcopy for some copies.

This API was superseded by the [PutBlockFromURL](https://learn.microsoft.com/en-us/rest/api/storageservices/put-block-from-url) / [PutBlockList](https://learn.microsoft.com/en-us/rest/api/storageservices/put-block-list) APIs. This API was introduced in 2018-03-28 which was after the rclone azureblob backend was created in 2017-07-25!

Updating to using this API would allow rclone to do the server side copy in chunks - as a multi-part server side copy - and most importantly allow these chunks to run in parallel. This is used by [AzCopy](https://github.com/Azure/azure-storage-azcopy) and other tools.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
Here is some test code for this - testing appreciated:
    
This implements multipart server side copy to improve copying from one
azure region to another by orders of magnitude (from 30s for a 100M
file to 10s for a 10G file with --azureblob-upload-concurrency 500).
    
- Add `--azureblob-copy-cutoff` to control the cutoff from single to multipart copy
- Add `--azureblob-copy-concurrency` to control the copy concurrency
- Add ServerSideAcrossConfigs flag as this now works properly
- Implement multipart copy using put block list API
- Shortcut multipart copy for same storage account
- Override with `--azureblob-use-copy-blob`

[v1.69.0-beta.8471.f67ac4d38.fix-8249-azure-blob-server-side-copy](https://beta.rclone.org/branch/fix-8249-azure-blob-server-side-copy/v1.69.0-beta.8471.f67ac4d38.fix-8249-azure-blob-server-side-copy/) on branch [fix-8249-azure-blob-server-side-copy](https://github.com/rclone/rclone/tree/fix-8249-azure-blob-server-side-copy) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.70
